server/graphql/asset/src/types.rs

```diff
@@ -153,6 +174,16 @@ impl AssetNode {
             .map(AssetCatalogueItemNode::from_domain))
     }
 
+    pub async fn locations(&self, ctx: &Context<'_>) -> Result<AssetLocationConnector> {
```

Comment:
```suggestion
    pub async fn locations(&self, ctx: &Context<'_>) -> Result<LocationConnector> {
```

I think you can re-use the LocationConnector type, don't need the AssetLocationConnector at all!